mslice/workspace/histo_mixin.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from mantid.simpleapi import CreateMDHistoWorkspace, ReplicateMD
+from mantid.api import AlgorithmManager
+
+
+class HistoMixin(object):
+
+    def get_signal(self):
+        """Gets data values (Y axis) from the workspace as a numpy array. Overrides Workspace method."""
+        return self._raw_ws.getSignalArray()
+
+    def get_error(self):
+        """Gets error values (E) from the workspace as a numpy array. Overrides Workspace method."""
+        return np.sqrt(self.get_variance())
+
+    def get_variance(self):
+        """Gets variance (error^2) from the workspace as a numpy array. Overrides Workspace method"""
+        return self._raw_ws.getErrorSquaredArray()
+
+    def _binary_op_array(self, operator, other):
+        """
+        Perform binary operation (+,-,*,/) using a 1D numpy array.
+
+        CreateMDHistoWorkspace using numpy array and then use ReplicateMD so it matches the shape of _raw_ws.
+        Then, apply operator.
+
+        :param operator: binary operator to apply (add/sub/mul/div)
+        :param other: 1D numpy array to use with operator.
+        :return: new HistogramWorkspace
+        """
+        min = np.amin(other)
+        max = np.amax(other)
+        size = other.size
+        ws = CreateMDHistoWorkspace(Dimensionality=1, Extents='' + str(min) + ',' + str(max),
+                                    SignalInput=other, ErrorInput=other, NumberOfBins=str(size),
+                                    Names=self._raw_ws.getDimension(0).getName(), Units='MomentumTransfer')
+        try:
+            replicated = ReplicateMD(self._raw_ws, ws)
+            return operator(self._raw_ws, replicated)
+        except RuntimeError:
+            raise RuntimeError("List or array must have same number of elements as an axis of the workspace")
+
+    def __pow__(self, exponent):
+        pow_alg = AlgorithmManager.createUnmanaged('PowerMD')
```

Comment:
Can we do the same with `CreateMDHistoWorkspace` and `ReplicateMD` above? Maybe a utility function for running an  algorithm in this way that accepts `(name, *args, **kwargs)`?

Reply:
@martyngigg This seems like something that would be useful in Mantid also - this is a common pattern for running algorithms but it's quite verbose. Maybe it would be better to implement something in Mantid that Mslice could import.

Reply:
@MatthewBowles Agreed. @gvardany is working on something now where you would be able to do `ws = PowerMD(..., RunAsChild=True)` that would achieve the same result. I don't think we should wait for that.